graphql_compiler/macros/__init__.py

```diff
@@ -167,16 +172,30 @@ def get_schema_for_macro_definition(schema):
     """Returns a schema with macro directives.
 
     This returned schema can be used to validate macro definitions, and support GraphQL
-    macro editors, enabling them to autocomplete one the @macro_edge_definition and
-    @macro_edge_target directives.
+    macro editors, enabling them to autocomplete on the @macro_edge_definition and
+    @macro_edge_target directives. Some directives that are disallowed in macro edge definitions,
+    like @output and @output_source, will be removed from the directives list.
+    Raises an error if any non-default directives are present in the given schema.
 
     Args:
         schema: GraphQLSchema over which we want to write macros
 
     Returns:
         GraphQLSchema usable for writing macros
     """
-    raise NotImplementedError()  # TODO(bojanserafimov): Implement
+
+    macro_definition_schema = copy(schema)
+    macro_definition_schema_directives = schema.get_directives()
+    _check_for_nondefault_macro_directives(macro_definition_schema_directives)
+    macro_definition_schema_directives = include_required_macro_directives(
+        macro_definition_schema_directives)
+    macro_definition_schema_directives = exclude_disallowed_directives_in_macros(
```

Comment:
Why are we excluding disallowed directives? Didn't Bojan and I agree on just asserting that we simply raise an error if any of schema directives is not allowed?

Reply:
Some default directives, like @output and @output_source are disallowed. However, if we raised errors, we wouldn't be able to use @output and macro edges in the same schema.